brion/plugin/compartmentReportHDF5.cpp

```diff
@@ -637,16 +757,123 @@ void CompartmentReportHDF5::_allocateDataSet()
         std::vector<size_t>{frames, _targetMapping.frameSize});
 
     HighFive::DataSetCreateProps chunking;
-    auto chunkDims = _computeChunkDims(_targetMapping.cellSizes, 5);
+    auto chunkDims =
+        _computeChunkDims(_targetMapping.cellSizes, _cellsToFramesRatio);
     if (chunkDims[0] >= frames)
-        // Only apply chunking if the dataset is large enough. Otherwise HDF5
-        // gives an error.
-        chunking.add(HighFive::Chunking(chunkDims));
+        chunkDims[0] = frames;
+    if (chunkDims[1] > _targetMapping.frameSize)
+        chunkDims[1] = _targetMapping.frameSize;
+    chunking.add(HighFive::Chunking(chunkDims));
+
+    HighFive::DataSetAccessProps caching;
+    size_t blocksPerFrame =
+        (_targetMapping.frameSize + chunkDims[1] - 1) / chunkDims[1];
+    caching.add(
+        HighFive::Caching(blocksPerFrame * _blockSize, blocksPerFrame + 1));
 
     _data.reset(new HighFive::DataSet(
-        _file->createDataSet<float>("data", dataspace, chunking)));
+        _file->createDataSet<float>("data", dataspace, chunking, caching)));
 
     detail::addStringAttribute(*_data, "dunit", _dunit);
 }
+
+void CompartmentReportHDF5::_parseWriteOptions(const URI& uri)
+{
+    for (auto i = uri.queryBegin(); i != uri.queryEnd(); ++i)
```

Comment:
const auto&

Reply:
In this case it doesn't matter, but since this is just an iterator, don't you think that adding a reference would make the code slower for a long loop? Even with optimizations I have the feeling that the code would have more pointer chasing with the reference.